worktree.go

```diff
@@ -140,6 +140,13 @@ func (w *Worktree) Checkout(opts *CheckoutOptions) error {
 		return err
 	}
 
+	c, err := w.getCommitFromCheckoutOptions(opts)
+	if err == plumbing.ErrReferenceNotFound {
```

Comment:
I think we also want to check `opts.Create` here. If false, I think we want to continue to error out.

```
$ git init
$ cat .git/HEAD
ref: refs/heads/master
$ cat .git/refs/heads/master
cat: .git/refs/heads/master: No such file or directory
$ git checkout aBranch
error: pathspec 'aBranch' did not match any file(s) known to git.
$ cat .git/HEAD
ref: refs/heads/master
$ git checkout -b aBranch
Switched to a new branch 'aBranch'
$ cat .git/HEAD
ref: refs/heads/aBranch
$ cat .git/refs/heads/aBranch
cat: .git/refs/heads/aBranch: No such file or directory
```